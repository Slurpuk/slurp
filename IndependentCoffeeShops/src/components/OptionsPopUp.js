import CheckboxSectionList from './CheckboxSectionList';
import React from 'react';
import {Dimensions, StyleSheet, Text, View} from 'react-native';
import textStyles from '../../stylesheets/textStyles';
import PrimaryButton from '../../PrimaryButton';

const OptionsPopUp = ({data, renderer, product_name, curr_price}) => {
  return (
    <View style={styles.container}>
      <Text style={[textStyles.headingOne, styles.product_name]}>
        {product_name}
      </Text>
      <View style={styles.list}>
        <CheckboxSectionList DATA={data} renderItem={renderer} />
      </View>
      <View style={styles.button}>
        <PrimaryButton text={`Add To Order  Â£${curr_price + '0'}`} />
      </View>
    </View>
  );
};

const screenWidth = Dimensions.get('window').width;
const screenHeight = Dimensions.get('window').height;
const styles = StyleSheet.create({
  container: {
    paddingLeft: '3%',
    width: 0.9128 * screenWidth,
    height: 0.5723 * screenHeight,
    backgroundColor: 'white',
    paddingVertical: '4%',
    position: 'relative',
  },

  product_name: {
    color: 'black',
    marginLeft: '2%',
    marginBottom: '1%',
  },

  list: {
    flex: 1,
    height: '60%',
  },

  button: {
    marginLeft: '-3%',
  },
});

export default OptionsPopUp;
