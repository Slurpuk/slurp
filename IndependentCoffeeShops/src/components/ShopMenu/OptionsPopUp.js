import CheckboxSectionList from './CheckboxSectionList';
import React, {useState} from 'react';
import {Dimensions, StyleSheet, Text, View} from 'react-native';
import textStyles from '../../../stylesheets/textStyles';
import PrimaryButton from '../../SubComponents/PrimaryButton';

const OptionsPopUp = ({data, renderer, product_name, curr_price}) => {
  const [totalPrice, setTotalPrice] = useState(curr_price); // Current total price in pennies
  const [options, setOptions] = useState({}); // List of options currently selected

  const updateOptions = (name, price, isAdd) => {
    if (isAdd) {
      setTotalPrice(price + totalPrice);
      setOptions(prevState => ({
        ...prevState,
        [name]: price,
      }));
    } else {
      setTotalPrice(totalPrice - price);
      let newState = options;
      delete newState[name];
      setOptions(newState);
    }
  };

  return (
    <View style={styles.container}>
      <Text style={[textStyles.headingOne, styles.product_name]}>
        {product_name}
      </Text>
      <View style={styles.list}>
        <CheckboxSectionList
          updateOptions={updateOptions}
          DATA={data}
          renderItem={renderer}
        />
      </View>
      <View style={styles.button}>
        <PrimaryButton
          text={`Add To Order  Â£${(totalPrice / 100).toPrecision(3)}`}
        />
      </View>
    </View>
  );
};

const screenWidth = Dimensions.get('window').width;
const screenHeight = Dimensions.get('window').height;
const styles = StyleSheet.create({
  container: {
    paddingLeft: '3%',
    width: 0.9128 * screenWidth,
    height: 0.5723 * screenHeight,
    backgroundColor: 'white',
    paddingVertical: '4%',
    position: 'relative',
  },

  product_name: {
    color: 'black',
    marginLeft: '2%',
    marginBottom: '1%',
  },

  list: {
    flex: 1,
    height: '60%',
  },

  button: {
    marginLeft: '-3%',
  },
});

export default OptionsPopUp;
