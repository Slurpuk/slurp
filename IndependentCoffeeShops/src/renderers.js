import React from 'react';
import Option from './components/Option';

const renderOption = ({item, updateOptions}) => (
  <Option
    name={item.name}
    price={item.price}
    currency={item.currency}
    updateOptions={updateOptions}
  />
);

export default renderOption;
