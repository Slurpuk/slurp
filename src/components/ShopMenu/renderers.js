import React from 'react';
import Option from './Option';
import MenuItem from './MenuItem';

/**
 * Renderer for rendering item options, reusable throughout the app.
 * @param item The option to display
 * @param updateOptions The setState method for updating the current list of options selected.
 */
const renderOption = ({item, updateOptions}) => (
  <Option option={item} updateOptions={updateOptions} />
);

/**
 * Renderer for rendering menu items, reusable throughout the app.
 * @param item The option to display
 */
const renderMenuItem = ({item}) => <MenuItem item={item} />;

export default {
  renderOption,
  renderMenuItem,
};
