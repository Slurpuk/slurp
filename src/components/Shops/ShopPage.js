import React, {Component} from 'react';
import ShopIntro from './ShopIntro';
import {StyleSheet, Text, View} from 'react-native';
import CustomButton from '../../sub-components/CustomButton';
import Menu from '../ShopMenu/Menu';

const ShopPage = ({
  shopName,
  shopIntroText,
  DATA,
  renderItem,
  renderSection,
}) => {
  return (
    <>
      <ShopIntro
        shopName={shopName}
        shopIntroText={shopIntroText}
        likeness={'65%'}
        timeToOrder={7}
      />
      <Menu DATA={DATA} renderItem={renderItem} renderSection={renderSection} />
      <View />
    </>
  );
};

const shopPageCustom = StyleSheet.create({
  absoluteArea: {
    height: 60,
    backgroundColor: '',
    justifyContent: 'center',
    alignItems: 'center',
    position: 'relative',
  },
});
export default ShopPage;
