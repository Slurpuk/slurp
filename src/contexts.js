import React from 'react';
import {Animated} from 'react-native';

export const defaultGlobalContextValues = {
  isLoadingShops: true,
  isLoadingUser: true,
  currentUser: null,
  currentUserRef: null,
  currentShopKey: '',
  currentShop: null,
  pastOrders: [],
  currentOrders: [],
  listOfShops: [],
  currentBasket: [],
  isShopIntro: false, // Is the shop page bottom sheet up.
  isFirstTime: true, // Is it the first time the app is downloaded
  adaptiveOpacity: new Animated.Value(0), // Animation fading value.
  locationIsEnabled: false,
  isConnected: false,
};

export const defaultMapContextValues = {
  mapCenter: {
    longitudeDelta: 0.01,
    latitudeDelta: 0.01,
  },
  isSearchBarFocused: false,
  isUserCentered: false,
  isManuallyCentered: false,
};
const GlobalContext = React.createContext(defaultGlobalContextValues);

const MapContext = React.createContext(defaultMapContextValues);

export {GlobalContext, MapContext};
