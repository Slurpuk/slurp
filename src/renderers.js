import React from 'react';
import Option from './components/ShopMenu/Option';
import MenuSection from './components/ShopMenu/MenuSection';
import ItemCard from './components/ShopMenu/ItemCard';
import ShopCard from './components/Shops/ShopCard';
import ShopList from './components/Shops/ShopList';
import ShopIntro from './components/Shops/ShopIntro';
import Menu from './components/ShopMenu/Menu';
import ItemsData from './fake-data/ItemsData';

const renderOption = ({item, updateOptions}) => (
  <Option
    name={item.name}
    price={item.price}
    currency={item.currency}
    updateOptions={updateOptions}
  />
);

const renderMenu = menu => menu;

const renderMenuSection = ({item, renderItem}) => (
  <MenuSection section={item} renderItem={renderItem} />
);

const renderItemCard = ({item}) => <ItemCard item={item} />;

// const renderShopCard = ({item}) => (
//   <ShopCard
//     name={item.name}
//     likeness={item.likeness}
//     queue={item.queue}
//     image={item.image}
//   />
// );
// const renderShopList = ({item, renderItem}) => (
//   <ShopList DATA={item.data} renderItem={renderItem} />
// );

export default {
  renderOption,
  renderMenuSection,
  renderItemCard,
  // renderShopCard,
  // renderShopList,
  renderMenu,
};
