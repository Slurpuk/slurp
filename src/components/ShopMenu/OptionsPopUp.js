import CheckboxSectionList from './CheckboxSectionList';
import React, {useContext, useEffect, useState} from 'react';
import {
  Dimensions,
  StyleSheet,
  Text,
  TouchableHighlight,
  View,
} from 'react-native';
import textStyles from '../../../stylesheets/textStyles';
import CustomButton from '../../sub-components/CustomButton';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import {ShopContext} from '../Shops/ShopPage';

const OptionsPopUp = ({data, renderer, product_name, curr_price}) => {
  const context = useContext(ShopContext);
  const [totalPrice, setTotalPrice] = useState(curr_price); // Current total price in pennies
  const [options, setOptions] = useState({}); // List of options currently selected
  const [isVisible, setVisible] = useState(true); // State for tracking if the popup is currently visible
  const updateOptions = (name, price, isAdd) => {
    if (isAdd) {
      setTotalPrice(price + totalPrice);
      setOptions(prevState => ({
        ...prevState,
        [name]: price,
      }));
      console.log(options);
    } else {
      setTotalPrice(totalPrice - price);
      let newState = options;
      delete newState[name];
      setOptions(newState);
    }
  };

  useEffect(() => {
    if (isVisible === false) {
      context.setOptionsVisible(isVisible);
    }
  }, [isVisible]);

  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <Text style={[textStyles.headingOne, styles.product_name]}>
          {product_name}
        </Text>
        <TouchableHighlight
          style={styles.icon}
          underlayColor={'white'}
          onPress={() => setVisible(false)}
        >
          <Icon size={30} color="black" name="close" />
        </TouchableHighlight>
      </View>
      <View style={styles.list}>
        <CheckboxSectionList
          updateOptions={updateOptions}
          DATA={data}
          renderItem={renderer}
        />
      </View>
      <CustomButton
        text={`Add To Order  Â£${(totalPrice / 100).toPrecision(3)}`}
        priority={'primary'}
        width={screenWidth * 0.79}
        onPress={() => setVisible(false)}
      />
    </View>
  );
};

const screenWidth = Dimensions.get('window').width;
const screenHeight = Dimensions.get('window').height;
const styles = StyleSheet.create({
  container: {
    display: 'flex',
    justifyContent: 'center',
    paddingLeft: '7%',
    width: 0.9128 * screenWidth,
    height: 0.5723 * screenHeight,
    backgroundColor: 'white',
    paddingVertical: '6%',
    position: 'absolute',
    top: '20%',
    bottom: '23%',
    left: '4%',
    right: '4%',
    elevation: 200,
    zIndex: 100,
  },

  header: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },

  product_name: {
    color: 'black',
    marginLeft: '2%',
  },

  list: {
    flex: 1,
    height: '60%',
  },

  icon: {
    marginRight: '7%',
    marginBottom: '2%',
  },
});

export default OptionsPopUp;
