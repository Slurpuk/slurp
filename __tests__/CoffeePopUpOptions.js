import {render} from '@testing-library/react-native';
import React from 'react';
import {GlobalContext} from '../src/contexts';
import {Alert} from 'react-native';
import OptionsPopUp from '../src/components/ShopMenu/OptionsPopUp';
import renderers from '../src/components/ShopMenu/renderers';
import {ShopContext} from '../src/screens/ShopPage';
import {getByTestId} from '@testing-library/react';

describe('Coffee Pop Up', function () {
  const spyAlert = jest
    .spyOn(Alert, 'alert')
    //@ts-ignore
    .mockImplementation((title, message, callbackOrButtons) =>
      callbackOrButtons[1].onPress(),
    );
  afterEach(() => spyAlert.mockClear());

  const shopContextMock = {
    setOptionsVisible: jest.fn(),
    setCurrItem: jest.fn(),
    menuData: {
      defaultMilk: 'Dairy',
    },
  };

  describe('coffee options', function () {
    it('should sign up a user', function () {
      const globalContextMock = {
        isFirstTime: false,
      };
      const {getByTestId} = render(
        <GlobalContext.Provider value={globalContextMock}>
          <ShopContext.Provider value={shopContextMock}>
            <OptionsPopUp
              data={[]}
              item={{
                name: 'Latte',
                price: 2.5,
                quantity: 2,
                has_options: false,
              }}
              renderer={renderers.renderOption}
            />
          </ShopContext.Provider>
        </GlobalContext.Provider>,
      );
      expect(getByTestId('options_pop_up')).toBeTruthy();
    });
  });
});
