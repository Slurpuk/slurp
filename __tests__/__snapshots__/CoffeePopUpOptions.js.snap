// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coffee Pop Up coffee options should sign up a user 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "bottom": "23%",
      "display": "flex",
      "elevation": 200,
      "height": 800.4,
      "justifyContent": "center",
      "left": "4%",
      "paddingLeft": "7%",
      "paddingVertical": "6%",
      "position": "absolute",
      "right": "4%",
      "top": "20%",
      "width": 684.5999999999999,
      "zIndex": 100,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontFamily": "JosefinSans-Bold",
            "fontSize": 25,
            "includeFontPadding": false,
            "letterSpacing": 0.5,
            "textTransform": "uppercase",
          },
          Object {
            "color": "black",
            "marginLeft": "2%",
          },
        ]
      }
    >
      Latte
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginBottom": "2%",
          "marginRight": "7%",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": undefined,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        󰅖
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "height": "60%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": "1.4%",
          "paddingBottom": "4%",
          "paddingTop": "0%",
        }
      }
    >
      <RCTScrollView
        data={Array []}
        getItem={[Function]}
        getItemCount={[Function]}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
      >
        <View />
      </RCTScrollView>
    </View>
  </View>
  <TextInput
    testID="foo"
  >
    <View>
      <View
        collapsable={false}
        style={
          Object {
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#087562",
              "borderRadius": 13,
              "display": "flex",
              "flexDirection": "row",
              "height": 43,
              "justifyContent": "center",
              "opacity": 1,
              "width": 592.5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#EFEFEF",
                "fontFamily": "Poppins-SemiBold",
                "fontSize": 17,
                "letterSpacing": 0.5,
              }
            }
          >
            Add To Order  £2.50
          </Text>
        </View>
      </View>
    </View>
  </TextInput>
</View>
`;

exports[`Coffee Pop Up should add a milk 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "bottom": "23%",
      "display": "flex",
      "elevation": 200,
      "height": 800.4,
      "justifyContent": "center",
      "left": "4%",
      "paddingLeft": "7%",
      "paddingVertical": "6%",
      "position": "absolute",
      "right": "4%",
      "top": "20%",
      "width": 684.5999999999999,
      "zIndex": 100,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontFamily": "JosefinSans-Bold",
            "fontSize": 25,
            "includeFontPadding": false,
            "letterSpacing": 0.5,
            "textTransform": "uppercase",
          },
          Object {
            "color": "black",
            "marginLeft": "2%",
          },
        ]
      }
    >
      Latte
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginBottom": "2%",
          "marginRight": "7%",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": undefined,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        󰅖
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "height": "60%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": "1.4%",
          "paddingBottom": "4%",
          "paddingTop": "0%",
        }
      }
    >
      <RCTScrollView
        data={Array []}
        getItem={[Function]}
        getItemCount={[Function]}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
      >
        <View />
      </RCTScrollView>
    </View>
  </View>
  <View>
    <View
      collapsable={false}
      style={
        Object {
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#087562",
            "borderRadius": 13,
            "display": "flex",
            "flexDirection": "row",
            "height": 43,
            "justifyContent": "center",
            "opacity": 1,
            "width": 592.5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#EFEFEF",
              "fontFamily": "Poppins-SemiBold",
              "fontSize": 17,
              "letterSpacing": 0.5,
            }
          }
        >
          Add To Order  £2.50
        </Text>
      </View>
    </View>
  </View>
</View>
`;
